autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -41,9 +41,21 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
         LoggingUtils.buildMdcContext(cluster);
 
         StackStatusV4Response statusResponse = cloudbreakCommunicator.getStackStatusByCrn(cluster.getStackCrn());
-        boolean clusterAvailable = Optional.ofNullable(statusResponse.getStatus()).map(Status::isAvailable).orElse(false)
-                && Optional.ofNullable(statusResponse.getClusterStatus()).map(Status::isAvailable).orElse(false);
-        LOGGER.debug("Analysing CBCluster Status '{}' for Cluster '{}' ", statusResponse, cluster.getStackCrn());
+
+        boolean clusterAvailable;
+        boolean clusterWithStoppedNodes = false;
+        if (cluster.isStopStartScalingEnabled()) {
+            clusterAvailable = Optional.ofNullable(statusResponse.getStatus()).map(Status::isAvailable).orElse(false);
+            clusterWithStoppedNodes =
+                    Optional.ofNullable(statusResponse.getStatus()).map(s -> s == Status.AVAILABLE_WITH_STOPPED_INSTANCES).orElse(false);
```

Comment:
AVAILABLE_WITH_STOPPED_INSTANCES was explicitly introduced to make sure that existing operations, which rely on the AVAILABLE state, cannot be performed when the cluster is in this state. AVAILABLE_WITH_STOPPED_ISNSTANCES ends up being referred to by operations which we want to allow.
This is the simples way to ensure this, and make sure current operations continue to behave properly. The design doc lists out the various operations and the expectations from them.

Reply:
What happens if the customer somehow starts those stopped (scaled down by stopping) nodes on the cloud provider side? How do we handle this scenario?

Reply:
I think it would be better to only use the `AVAILABLE` state and introduce or reuse old one in the `InstanceStatus` enum. When an operation request comes in a validation logic can figure out if this is allowed or not.

Reply:
> What happens if the customer somehow starts those stopped (scaled down by stopping) nodes on the cloud provider side? How do we handle this scenario?
Today, the customer will end up with started instances, and services not running on these instances. They're not supposed to manipulate nodes on the cloud provider other than to resolve other issues.
That said, this is also one of the recovery modes. Start the instances on the cloud provider. Go to CM and provision via the UI - and you have a functional cluster again. (Think of the scenario where for some other reason a regular upscale is failing).
In Phase2, which focuses more on better resiliency - such instances will be considered as part of start operations, where we take care of starting the services via CM.

> I think it would be better to only use the AVAILABLE state and introduce or reuse old one in the InstanceStatus enum. When an operation request comes in a validation logic can figure out if this is allowed or not.
The intention of the new state is to make sure the feature does not interfere with operations for existing clusters. Minimize the risk of having to hunt down all operations, and change logic for them.

Reply:
AVAILABLE_WITH_STOPPED_INSTANCES -> it means the cluster is available but we have stopped nodes. It is denormalization and you can figure out the same if you check state is AVAILABLE then check the states of the instances (Any of them stopped). @akanto tried to get rid of meaningless statuses and he deleted some of them in the near past, we don't want to introduce states like this.